tests/asserts/keywords.py

```diff
@@ -718,6 +743,23 @@ def process_strats(src, paramData):
                                 msg = f"🔴ERROR: Strat ID {strat_id} is not less than nextStratId ({next_strat_id}):{stratRef}"
                                 messages["reds"].append(msg)
                                 messages["counts"]["reds"] += 1
+                        if heated and not check_heat_req({"and": strat["requires"]}):
+                            if fromNode == toNode and "leaveWithRunway" in strat.get("exitCondition", []):
+                                # Ok since there is implicit heat frames in leavesWithRunway, and it is normal
+                                # if no explicit ones to be present for a strat going from the door to itself.
+                                pass
+                            elif "comeInWithGMode" in strat.get("entranceCondition", []) or "gModeRegainMobility" in strat:
```

Comment:
A gmode strat off of a junction probably will have heat frames when you end gmode.  Junction to junction isn't likely.

There could be a case for Gmode->Ridley->Leave through open door to save the boss flag.  GT too.  ~~But I've also heard the flag is set when drops appear and that would take releasing GMode too 🤔~~ 
I don't know how you do the fight damageless, but you keep the flag on boss kill.  There's also time to end gmode and exit the room before the (open)door closes if you had to do it that way.

Reply:
Yeah, it's a good point, the "comeInWithGMode" exception is a little too broad since some of these strats should be having heat frames. I changed it now to carve out the G-mode exceptions more narrowly, only applying now in these cases:
1. Strat has both "comeInWithGMode" and "leaveWithGMode", so it's spending the entire time in G-mode.
2. Strat has "gModeRegainMobility", which also will be spending the entire time in G-mode.
This didn't cause any new strats to fail the test. I think this is expected since the G-mode applications in heated rooms have not yet been done.

Reply:
I dont think there are any g-mode usage strats in heated rooms yet.

It could also be fine to have strats that come in with g-mode and end at a item node or a g-mode junction, but it's fine just manually adding 0 frames to those as well - upper Norfair rooms are usually much simpler, so I dont think there will be a lot of g-mode junctions, and there arent many items in heated rooms.